### PR TITLE
Added implementation and test-cases to demonstrate Codable routing.

### PR DESCRIPTION
**Pull request by mamabusi**

### CODE REVIEW

File: Sources/KituraSampleRouter/RouterCreator.swift [hunk 1]
```diff
@@ -295,3 +315,13 @@ public struct RouterCreator {
         return router
     }
 }
+
+func persistBookHandler(book: Book, completion: (Book?, RequestError?) -> Void ) {
+    bookStore[book.name] = book
+    completion(bookStore[book.name], nil)
+}
+
+func fetchBooksHandler(completion: ([Book]?, RequestError?) -> Void ) {
```
Comment (KyeMaloy97) [inline Sources/KituraSampleRouter/RouterCreator.swift]: This function doesn't appear to be used anywhere?

File: Sources/KituraSampleRouter/Book.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    
+    init?(name: String, author: String, rating: Int) {
+        
+        // The name must not be empty
+        guard !name.isEmpty else {
+            return nil
+        }
+        
+        // The rating must be between 0 and 5 inclusively
+        guard (rating >= 0) && (rating <= 5) else {
```
Comment (KyeMaloy97) [inline Sources/KituraSampleRouter/Book.swift]: `guard 0...5 ~= rating else {` might look more concise

**Comment by mamabusi** [comment]
@KyeMaloy97 I have addressed the changes requested. 

As to the failures seen, I cloned the master branch of Kitura-sample and see the same errors happening, without my code changes.

**Comment by mamabusi** [comment]
@KyeMaloy97 Review changes have been addressed and the home page modified too. Could you please take a look. Thanks.

**Comment by KyeMaloy97** [comment]
All seems good to me, and running and passing locally on my machine.

**Pull request merged**